server/src/main/java/com/unosquare/carmigo/controller/AdminUserController.java

```diff
@@ -0,0 +1,92 @@
+package com.unosquare.carmigo.controller;
+
+import com.github.fge.jsonpatch.JsonPatch;
+import com.unosquare.carmigo.model.request.CreateDriverViewModel;
+import com.unosquare.carmigo.model.response.DriverViewModel;
+import com.unosquare.carmigo.model.response.PassengerViewModel;
+import com.unosquare.carmigo.model.response.PlatformUserViewModel;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/v1/admin/users")
+@Tag(name = "Admin User Controller")
+public class AdminUserController {
```

Comment:
Consider the need of an admin controller or should we lock endpoints in particular controllers?

It means entity management is all in one place and granular access is set at an earlier stage.  e.g what would happen if you wanted to create an endpoint in the admin controller that was only accessed by another type of role?

In general, wildcard access is something that needs to be very well considered.

Reply:
AdminUserController was removed. Using @PreAuthorize annotations instead